pinot-query-planner/src/main/java/org/apache/pinot/query/planner/explain/ExplainNodeSimplifier.java

```diff
@@ -0,0 +1,169 @@
+
+
+public class ExplainNodeSimplifier {
+  private static final Logger LOGGER = LoggerFactory.getLogger(ExplainNodeSimplifier.class);
+
+  private ExplainNodeSimplifier() {
+  }
+
+  public static PlanNode simplifyNode(PlanNode plan1) {
+    Visitor planNodeMerger = new Visitor();
+    return plan1.visit(planNodeMerger, null);
+  }
+
+  private static class Visitor implements PlanNodeVisitor<PlanNode, Void> {
+    private static final String REPEAT_ATTRIBUTE_KEY = "repeatedOnSegments";
+
+    private PlanNode defaultNode(PlanNode node) {
+      List<PlanNode> inputs = node.getInputs();
+      List<PlanNode> newInputs = simplifyChildren(inputs);
+      return inputs != newInputs ? node.withInputs(newInputs) : node;
+    }
+
+    @Override
+    public PlanNode visitAggregate(AggregateNode node, Void context) {
+      return defaultNode(node);
+    }
+
+    @Override
+    public PlanNode visitFilter(FilterNode node, Void context) {
+      return defaultNode(node);
+    }
+
+    @Override
+    public PlanNode visitJoin(JoinNode node, Void context) {
+      return defaultNode(node);
+    }
+
+    @Override
+    public PlanNode visitMailboxReceive(MailboxReceiveNode node, Void context) {
+      return defaultNode(node);
+    }
+
+    @Override
+    public PlanNode visitMailboxSend(MailboxSendNode node, Void context) {
+      return defaultNode(node);
+    }
+
+    @Override
+    public PlanNode visitProject(ProjectNode node, Void context) {
+      return defaultNode(node);
+    }
+
+    @Override
+    public PlanNode visitSort(SortNode node, Void context) {
+      return defaultNode(node);
+    }
+
+    @Override
+    public PlanNode visitTableScan(TableScanNode node, Void context) {
+      return defaultNode(node);
+    }
+
+    @Override
+    public PlanNode visitValue(ValueNode node, Void context) {
+      return defaultNode(node);
+    }
+
+    @Override
+    public PlanNode visitWindow(WindowNode node, Void context) {
+      return defaultNode(node);
+    }
+
+    @Override
+    public PlanNode visitSetOp(SetOpNode node, Void context) {
+      return defaultNode(node);
+    }
+
+    @Override
+    public PlanNode visitExchange(ExchangeNode node, Void context) {
+      return defaultNode(node);
+    }
+
+    @Override
+    public PlanNode visitExplained(ExplainedNode node, Void context) {
+      if (!node.getType().contains(ExplainPlanDataTableReducer.COMBINE) || node.getInputs().size() <= 1) {
+        return defaultNode(node);
+      }
+      List<PlanNode> simplifiedChildren = simplifyChildren(node.getInputs());
+      PlanNode child1 = simplifiedChildren.get(0);
+
+      for (int i = 1; i < simplifiedChildren.size(); i++) {
+        PlanNode child2 = simplifiedChildren.get(i);
+        PlanNode merged = PlanNodeMerger.mergePlans(child1, child2, false);
+        if (merged == null) {
+          LOGGER.info("Found unmergeable inputs on node of type {}: {} and {}", node, child1, child2);
+          assert false : "Unmergeable inputs found";
```

Comment:
We shouldn't reach this code. All nodes at this point should be mergeable. In case we end up reaching this code in production, it would be better to not fail. But during tests we should to make it easier to find the problem.